pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -532,7 +561,9 @@ func (cb *ClusterBuilder) buildLocalityLbEndpoints(proxyNetworkView map[network.
 			continue
 		}
 		// TODO(nmittler): Consider merging discoverability policy with cluster-local
-		if !instance.Endpoint.IsDiscoverableFromProxy(cb.proxy) {
+		// TODO(ramaraochavali): Find a better way here so that we do not have build proxy.
+		// Currently it works because we only determine discoverability only by cluster.
+		if !instance.Endpoint.IsDiscoverableFromProxy(&model.Proxy{Metadata: &model.NodeMetadata{ClusterID: istio_cluster.ID(cb.clusterID)}}) {
```

Comment:
@nmittler Is proxy mandatory for DiscoverableFromProxy? Today it is always based on cluster (or network?) ? Can we change this to string parameter instead of proxy or add a new struct encapsulating what we need instead of proxy for that function parameter?